test/tools/TestExe/TestExe.cs

```diff
@@ -52,6 +60,36 @@ private static void EchoArgs(string[] args)
             }
         }
 
+        // <Summary>
+        // Echos the raw command line received by the process plus the arguments passed in.
+        // </Summary>
+        private static void EchoCmdLine(string[] args)
+        {
+            string rawCmdLine = "N/A";
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                nint cmdLinePtr = Interop.GetCommandLineW();
+                rawCmdLine = Marshal.PtrToStringUni(cmdLinePtr);
+            }
+
+            Console.WriteLine(rawCmdLine);
```

Comment:
Would it make more sense to return a non-zero exit code and write message to StdErr?

Reply:
> return a non-zero exit code and write message to StdErr

What is the purpose of doing this? Do you have some testing scenarios in mind that would need it?

Reply:
It is what _any_ well-behaved CLI should do: print error messages to stderr, indicate failure via a non-zero exit code. Except in cases where backward compatibility must (unfortunately) be maintained, there's no good reason _not_ to do this.

Reply:
I agree with @mklement0, even though this is a test tool, we want to set good examples as other folks may end up using this code for their own projects.

Reply:
I'm afraid I need some clarification on this.
Are you suggesting returning non-zero exit code and write an error message instead of just writing out `N/A` when running on non-windows? The initial comment looks like suggesting always doing that.

Reply:
If I understood @SteveL-MSFT correctly, his comment was solely related to this _unrelated_ piece of code, which deals with passing an _unknown test name_ to the utility (the cited code _originally_ showed that).

```c#
                    default:
                        Console.WriteLine("Unknown test {0}. Run with '-h' for help.", args[0]);
                        break;
```

Reply:
@SteveL-MSFT Is that what you mean?

Reply:
I have updated that piece of code based on the feedback. Please let me know if you mean something else.